bin/generate_repos.py

```diff
@@ -31,17 +38,88 @@
     JSON = json.load(open_file)
 VCS_FOLDER = os.path.join(JSON['submitty_data_dir'], 'vcs', 'git')
 
+# =======================================================================
+def add_empty_commit(folder,which_branch):
+
+    assert (os.path.isdir(folder))
+    os.chdir(folder)
+
+    # check to see if there are any branches in the repo with commits
+    result = subprocess.run(['git', 'branch', '-v'], stdout=subprocess.PIPE)
+    s = result.stdout.decode('utf-8')
+    if s != "":
+        # do nothing if there is at least one branch with a commit
+        print('NOTE: this repo is non-empty (has a commit on at least one branch)')
+        return
+
+    # otherwise clone to a non-bare repo and add an empty commit
+    # to the specified branch
+    with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
did you mean to use this variable throughout instead of the hardcoded string `tmp`?

Reply:
thanks for that catch